File: src/games/stendhal/client/actions/RunicAltarAction.java
/***************************************************************************
 *                 (C) Copyright 2019-2024 - PolanieOnLine                 *
 ***************************************************************************
 ***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 ***************************************************************************/
package games.stendhal.client.actions;

import games.stendhal.client.gui.j2DClient;

class RunicAltarAction implements SlashAction {
	@Override
	public boolean execute(String[] params, String remainder) {
		j2DClient client = j2DClient.get();
		if (client == null) {
			return false;
		}

		client.getVisibleRunicAltar();
		return true;
	}

	@Override
	public int getMaximumParameters() {
		// TODO Auto-generated method stub
		return 0;
	}

	@Override
	public int getMinimumParameters() {
		// TODO Auto-generated method stub
		return 0;
	}
}

File: src/js/stendhal/action/RunicAltarAction.ts
/***************************************************************************
 *                   (C) Copyright 2026 - PolanieOnLine                    *
 ***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU Affero General Public License as        *
 *   published by the Free Software Foundation; either version 3 of the    *
 *   License, or (at your option) any later version.                       *
 *                                                                         *
 ***************************************************************************/

import { SlashAction } from "./SlashAction";

import { ui } from "../ui/UI";
import { UIComponentEnum } from "../ui/UIComponentEnum";


export class RunicAltarAction extends SlashAction {
	readonly minParams = 0;
	readonly maxParams = 0;

	override readonly desc = "Przełącza widoczność okna Ołtarza Runicznego.";


	execute(_type: string, _params: string[], _remainder: string): boolean {
		const runicAltar = ui.get(UIComponentEnum.RunicAltar);
		if (!runicAltar) {
			return false;
		}
		runicAltar.toggleVisibility();
		return true;
	}
}
